src/components/internal/FormFields/utils.ts

```diff
@@ -1,3 +1,27 @@
+import { JSX } from 'preact/jsx-runtime';
+
 const convertFullToHalf = (str: string) => str.replace(/[！-～]/g, r => String.fromCharCode(r.charCodeAt(0) - 0xfee0));
 
-export { convertFullToHalf };
+const filterDisallowedCharacters = (e: JSX.TargetedKeyboardEvent<HTMLInputElement>, inputType?: string) => {
+    if (inputType !== 'number') {
+        return;
+    }
+    const input = e.currentTarget as HTMLInputElement;
+    const { key } = e;
+
+    // Allow navigation and editing keys
+    const allowedKeys = ['Backspace', 'Delete', 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown', 'Tab'];
```

Comment:
I recommend obtaining these values from the globally defined `InteractionKeyCode` enum.

```suggestion
    const allowedKeys = [
        InteractionKeyCode.BACKSPACE,
        InteractionKeyCode.DELETE,
        InteractionKeyCode.ARROW_LEFT,
        InteractionKeyCode.ARROW_RIGHT,
        InteractionKeyCode.ARROW_UP,
        InteractionKeyCode.ARROW_DOWN,
        InteractionKeyCode.TAB,
    ];
```